dds/DCPS/RTPS/TypeLookup.h

```diff
@@ -0,0 +1,168 @@
+#ifndef OPENDDS_RTPS_TYPE_LOOKUP_H_
+#define OPENDDS_RTPS_TYPE_LOOKUP_H_
+
+/*
+ * Distributed under the OpenDDS License.
+ * See: http://www.opendds.org/license.html
+ */
+
+#include "dds/DCPS/XTypes/TypeObject.h"
+#include "RtpsRpcTypeSupportImpl.h"
+
+OPENDDS_BEGIN_VERSIONED_NAMESPACE_DECL
+
+namespace OpenDDS {
+  namespace XTypes {
+
+    const CORBA::ULong TypeLookup_getTypes_HashId = 25318099U;
+    const CORBA::ULong TypeLookup_getDependencies_HashId = 95091505U;
+
+    struct  TypeLookup_getTypes_In
+    {
+      OpenDDS::XTypes::TypeIdentifierSeq type_ids;
+
+      TypeLookup_getTypes_In() {}
+    };
+
+    struct  TypeLookup_getTypes_Out
+    {
+      OpenDDS::XTypes::TypeIdentifierTypeObjectPairSeq types;
+      OpenDDS::XTypes::TypeIdentifierPairSeq complete_to_minimal;
+
+      TypeLookup_getTypes_Out() {}
+    };
+
+    struct  TypeLookup_getTypes_Result
+    {
+      OpenDDS::XTypes::TypeLookup_getTypes_Out result;
+
+      TypeLookup_getTypes_Result() {};
+    };
+
+    typedef CORBA::Octet ContinuationPoint[32];
+
+    struct  TypeLookup_getTypeDependencies_In
+    {
+      OpenDDS::XTypes::TypeIdentifierSeq type_ids;
+      OpenDDS::XTypes::ContinuationPoint continuation_point;
+
+      TypeLookup_getTypeDependencies_In() {}
+    };
+
+    struct  TypeLookup_getTypeDependencies_Out
+    {
+      OpenDDS::XTypes::TypeIdentifierWithSizeSeq dependent_typeids;
+      OpenDDS::XTypes::ContinuationPoint continuation_point;
+
+      TypeLookup_getTypeDependencies_Out() {}
+    };
+
+    struct  TypeLookup_getTypeDependencies_Result
+    {
+      OpenDDS::XTypes::TypeLookup_getTypeDependencies_Out result;
+
+      TypeLookup_getTypeDependencies_Result() {};
+    };
+
+    // Call kinds
+    const TypeKind CK_NONE = 0x00;
```

Comment:
What does this relate to in the spec?

Reply:
Made it consistent with the spec.

Reply:
Seems like these constants should exist somewhere.